### PR TITLE
434: AuthorId added to entire temporal entity object tree

### LINKED ISSUE #434: get authorId working

### PR DESCRIPTION
**Pull request by jlhogan**
closes #434

### CODE REVIEW

File: entrypoint-test.sh [hunk 1]
```diff
@@ -1,14 +1,18 @@
...
 echo "moving test migrations into app..."
 testMigrations=$(ls test/HedwigTests/Fixtures/Migrations/*)
 cp test/HedwigTests/Fixtures/Migrations/* src/Hedwig/Migrations
 
-echo "installing dotnet ef utility..."
-until dotnet tool install --global dotnet-ef --version 3.0.0; do
-    sleep 1
-done
-export PATH="$PATH:/root/.dotnet/tools"
+if [ -z "$(dotnet tool list -g | grep dotnet-ef)" ]; then
```
Comment (cailyncodes) [inline entrypoint-test.sh]: Just curious: what does this do?

File: src/Hedwig/Data/HedwigContext.cs [hunk 2]
```diff
@@ -100,20 +101,32 @@ private bool IsTemporalEntityType<T>()
...
+      }
     }
 
     /// <summary>
     /// Gets the current UserId from HttpContext.User,
     /// or null if no UserId could be retrieved
     /// </summary>
     /// <returns></returns>
-    protected int? GetCurrentUserId()
+    protected virtual int? GetCurrentUserId()
```
Comment (cailyncodes) [inline src/Hedwig/Data/HedwigContext.cs]: Why the change here?

Comment (jlhogan) [reply src/Hedwig/Data/HedwigContext.cs]: for test :(

File: entrypoint-test.sh (hunk 1 above)
Comment (jlhogan) [reply entrypoint-test.sh]: running the test project on my local, i was running in infinite loop of trying to install the ef tool that alraedy existed. This makes it so it will only try to install the tool if it is not installed, which means i can run this on my local.

File: src/Hedwig/Data/HedwigContext.cs (hunk 2 above)
Comment (cailyncodes) [reply src/Hedwig/Data/HedwigContext.cs]: 😢
